src/main/java/com/coveros/selenified/OutputFile.java

```diff
@@ -186,12 +187,31 @@ public void setApp(App app) {
         this.app = app;
     }
 
+    /**
+     * Generates a unique filename, based on the package, class, method name, and invocation count
+     *
+     * @return String: the filename
+     */
+    private String generateFilename() {
+        String counter = "";
+        if (capabilities.getInstance() > 0) {
+            counter = "_" + capabilities.getInstance();
+        }
+        return test + counter + ".html";
+    }
+
     /**
      * Creates the directory and file to hold the test output file
      */
     private void setupFile() {
         if (!new File(directory).exists()) {
-            new File(directory).mkdirs();
+            if (!new File(directory).mkdirs()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1066)